src/applications/Application.py

```diff
@@ -75,13 +81,63 @@ def get_parameter_options(self) -> dict:
         """
         pass
 
+    def regenerate_on_iteration(self, config: dict) -> bool:
+        """Overwrite this to return True if the problem should be newly generated
+        on every iteration. Typically this will be the case if the problem is taken
+        from a statistical ensemble e.g. an erdos-renyi graph.
+        :param config:
```

Comment:
Können wir hier noch "the application configuration" hinzufügen @jusschwitalla ? Der Rest des PR schaut gut aus, vielen Dank!

Reply:
Hi Philipp, ja config wird potentiell benötigt, da die Entscheidung ob das Problem bei jeder Iteration neu generiert werden soll oder nicht von config abhäbgig sein kann.

Reply:
```suggestion
        :param config: the application configuration
```